src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,216 @@
+    private static final String INVALID_CERT_TYPE_ERROR = "Invalid certificate type.";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final CertificateManager certificateManager;
+    private final AtomicBoolean subscriptionResponseSent = new AtomicBoolean(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingEvent = new AtomicReference<>(null);
+    private final AuthorizationHandler authorizationHandler;
+    private KeyPair keyPair;
+    Consumer<X509Certificate> serverCertificateCallback = this::forwardServerCertificates;
+
+    /**
+     * Constructor.
+     *
+     * @param context               operation continuation handler
+     * @param certificateManager certificate manager
+     * @param authorizationHandler authorization handler
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         CertificateManager certificateManager,
+                                                         AuthorizationHandler authorizationHandler) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.certificateManager = certificateManager;
+        this.authorizationHandler = authorizationHandler;
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        certificateManager.unsubscribeFromServerCertificateUpdates(serverCertificateCallback);
+    }
+
+    /**
+     * Override to appropriately enforce stream responses are sent after the initial response.
+     * This only gets called if handleRequest returns normally and starts to send a response.
+     * <p>
+     * Here, we know that the initial SubscribeToCertificateUpdatesResponse response is sent. So, we send the
+     * first streaming response of CertificateUpdateEvent.
+     * </p>
+     */
+    @Override
+    public void afterHandleRequest() {
+        synchronized (firstStreamingEvent) {
+            if (subscriptionResponseSent.compareAndSet(false, true)) {
+                CertificateUpdateEvent event = firstStreamingEvent.get();
+                if (event != null) {
+                    sendStreamEvent(event);
+                }
+                firstStreamingEvent.set(null);
+            }
+        }
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
```

Comment:
nit: warn and error in the same line is confusing. Use `.kv("message", ...`?

Reply:
The message is an error.

Reply:
Then log at error? Warn and error together is confusing.

Reply:
I updated it to that based on this comment - https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/63#discussion_r855526347. It can be of error level IMO.

Reply:
The name of the constant is confusing, but this really should be logged at a warn level and not error. There is nothing broken about CDA functionality. It indicates a configuration issue elsewhere

Reply:
Thats why I suggested a name change.